File: js/src/maplestory/api/dto/character/characterVMatrix.ts
import {
  CharacterVMatrixBody,
  CharacterVMatrixCoreEquipmentDtoBody,
} from '../../response/character/characterVMatrixBody';

/**
 * 캐릭터 V매트릭스 정보
 */
export class CharacterVMatrixDto {
  /**
   * 조회 기준일
   */
  date: Date | null;

  /**
   * 캐릭터 직업
   */
  characterClass: string | null;

  /**
   * V코어 정보
   */
  characterVCoreEquipment: CharacterVMatrixCodeEquipmentDto[];

  /**
   * 캐릭터 잔여 매트릭스 강화 포인트
   */
  characterVMatrixRemainSlotUpgradePoint: number | null;

  constructor(obj: CharacterVMatrixBody) {
    const {
      date,
      character_class,
      character_v_core_equipment,
      character_v_matrix_remain_slot_upgrade_point,
    } = obj;

    this.date = date ? new Date(date) : null;
    this.characterClass = character_class;
    this.characterVCoreEquipment = character_v_core_equipment.map(
      (equipment) => new CharacterVMatrixCodeEquipmentDto(equipment),
    );
    this.characterVMatrixRemainSlotUpgradePoint =
      character_v_matrix_remain_slot_upgrade_point;
  }
}

/**
 * 캐릭터 V코어 정보
 */
export class CharacterVMatrixCodeEquipmentDto {
  /**
   * 슬롯 인덱스
   */
  slotId: string;

  /**
   * 슬롯 레벨
   */
  slotLevel: number;

  /**
   * 코어 명
   */
  vCoreName: string | null;

  /**
   * 코어 타입
   */
  vCoreType: string | null;

  /**
   * 코어 레벨
   */
  vCoreLevel: number;

  /**
   * 코어에 해당하는 스킬 명
   */
  vCoreSkill1: string;

  /**
   * (강화 코어인 경우) 코어에 해당하는 두 번째 스킬 명
   */
  vCoreSkill2: string | null;

  /**
   * (강화 코어인 경우) 코어에 해당하는 세 번째 스킬 명
   */
  vCoreSkill3: string | null;

  constructor(obj: CharacterVMatrixCoreEquipmentDtoBody) {
    const {
      slot_id,
      slot_level,
      v_core_name,
      v_core_type,
      v_core_level,
      v_core_skill_1,
      v_core_skill_2,
      v_core_skill_3,
    } = obj;

    this.slotId = slot_id;
    this.slotLevel = slot_level;
    this.vCoreName = v_core_name;
    this.vCoreType = v_core_type;
    this.vCoreLevel = v_core_level;
    this.vCoreSkill1 = v_core_skill_1;
    this.vCoreSkill2 = v_core_skill_2;
    this.vCoreSkill3 = v_core_skill_3;
  }
}

File: csharp/MapleStory.OpenAPI/Src/TMS/Dto/Character/CharacterVMatrix.cs
using Newtonsoft.Json;
using Base = MapleStory.OpenAPI.Common.DTO;

namespace MapleStory.OpenAPI.TMS.DTO
{
    /// <summary>
    /// 角色 V 矩陣欄位資訊與已裝備 V 核心資訊
    /// </summary>
    public class CharacterVMatrixDTO : Base.CharacterVMatrixDTO<CharacterVMatrixCoreEquipmentDTO>
    {
        /// <summary>
        /// 要搜尋的日期 (TST，每日資料中的小時與分鐘將顯示為 0)
        /// </summary>
        [JsonProperty("date")]
        public override DateTimeOffset? Date
        {
            get => _date?.ToOffset(TimeSpan.FromHours(8));
            set => _date = value;
        }

        private DateTimeOffset? _date;

        /// <summary>
        /// 角色職業
        /// </summary>
        [JsonProperty("character_class")]
        public override string? CharacterClass { get; set; }

        /// <summary>
        /// V 核心資訊
        /// </summary>
        [JsonProperty("character_v_core_equipment")]
        public override List<CharacterVMatrixCoreEquipmentDTO> CharacterVCoreEquipment { get; set; }

        /// <summary>
        /// 角色剩餘的矩陣強化點數
        /// </summary>
        [JsonProperty("character_v_matrix_remain_slot_upgrade_point")]
        public override int? CharacterVMatrixRemainSlotUpgradePoint { get; set; }
    }

    /// <summary>
    /// V 核心資訊
    /// </summary>
    public class CharacterVMatrixCoreEquipmentDTO : Base.CharacterVMatrixCoreEquipmentDTO
    {
        /// <summary>
        /// 欄位索引
        /// </summary>
        [JsonProperty("slot_id")]
        public override string SlotId { get; set; }

        /// <summary>
        /// 欄位等級
        /// </summary>
        [JsonProperty("slot_level")]
        public override long SlotLevel { get; set; }

        /// <summary>
        /// 核心名稱
        /// </summary>
        [JsonProperty("v_core_name")]
        public override string? VCoreName { get; set; }

        /// <summary>
        /// 核心類型
        /// </summary>
        [JsonProperty("v_core_type")]
        public override string? VCoreType { get; set; }

        /// <summary>
        /// 核心等級
        /// </summary>
        [JsonProperty("v_core_level")]
        public override long VCoreLevel { get; set; }

        /// <summary>
        /// 對應核心的技能名稱
        /// </summary>
        [JsonProperty("v_core_skill_1")]
        public override string VCoreSkill1 { get; set; }

        /// <summary>
        /// (若為強化核心) 對應核心的第二個技能名稱
        /// </summary>
        [JsonProperty("v_core_skill_2")]
        public override string? VCoreSkill2 { get; set; }

        /// <summary>
        /// (若為強化核心) 對應核心的第三個技能名稱
        /// </summary>
        [JsonProperty("v_core_skill_3")]
        public override string? VCoreSkill3 { get; set; }
    }
}
